src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Decrypt.cs

```diff
@@ -146,8 +146,40 @@ private static byte[] GetAsnSequenceWithContentNoValidation(ReadOnlySpan<byte> c
                 out Exception? exception)
             {
                 exception = null;
+
+#if NET
+                try
+                {
+                    using (SymmetricAlgorithm alg = OpenAlgorithm(contentEncryptionAlgorithm))
+                    {
+                        try
+                        {
+                            alg.Key = cek;
+                        }
+                        catch (CryptographicException ce)
+                        {
+                            throw new CryptographicException(SR.Cryptography_Cms_InvalidSymmetricKey, ce);
+                        }
+
+                        return alg.DecryptCbc(encryptedContent.Span, alg.IV);
+                    }
+                }
+                catch (CryptographicException ce)
+                {
+                    exception = ce;
+                    return null;
+                }
+                catch (NotSupportedException)
```

Comment:
One could argue that we may just not want to fall through here and this exception should be uncaught, hence the `Debug.Fail`, but since the code below exists, it seemed sensible to use it if somehow we end up in this situation.

Reply:
We can put all of the code below in `#else` instead of a fallback path by just copying up the "we got an internal implementation" assertion to right before calling the one-shot:

```C#
// All of the inbox implementations support one-shots, if we extend the library to support
// additional algorithm providers we might need a fallback plan.
Debug.Assert(alg.GetType().Assembly == typeof(Aes).Assembly);
```

Then this catch either throws UnreachableCodeException, or (probably better) just gets deleted.